app/components/Settings/scheduler/create-modal.jsx

```diff
@@ -0,0 +1,218 @@
+import Modal from '../../modal.jsx';
+import Error from '../../error.jsx';
+
+import { getHighlightMode } from '../../../constants/constants.js';
+
+import './create-modal.css';
+
+function noop() {}
+
+const FREQUENCIES = [
+    { label: 'Run every minute', value: 60 },
+    { label: 'Run every 5 minutes', value: 5 * 60 },
+    { label: 'Run hourly', value: 60 * 60 },
+    { label: 'Run daily', value: 24 * 60 * 60 },
+    { label: 'Run weekly', value: 7 * 24 * 60 * 60 }
+];
+
+const rowStyleOverride = { justifyContent: 'flex-start' };
+
+function generateFilename() {
+  let n = Math.floor(Math.random() * 1e8).toString();
+
+  // Pad 0 if needed
+  if (n.length < 8) {
+    n = `0${n}`;
+  }
+
+  return `Grid ${n}`;
+}
+
+export const FrequencySelector = props => (
+    <Select
+        options={FREQUENCIES}
+        value={props.value}
+        searchable={false}
+        onChange={props.onChange}
+    />
+);
+FrequencySelector.propTypes = {
+  value: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
+  onChange: PropTypes.func
+};
+
+// implements a modal window to schedule a new query
+class CreateModal extends Component {
+    static propTypes = {
+        initialCode: PropTypes.string,
+        // `initialFilename` currently unused but will be implemented once backend supports
+        initialFilename: PropTypes.string,
+        onClickAway: PropTypes.func.isRequired,
+        onSubmit: PropTypes.func.isRequired,
+        open: PropTypes.bool,
+        dialect: PropTypes.string
+    };
+    static defaultProps = {
+        initialCode: '',
+        initialFilename: '',
+        onClickAway: noop,
+        onSubmit: noop
+    };
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            code: props.initialCode,
+            filename: props.initialFilename || generateFilename(),
+            intervalType: null,
+            error: null,
+            loading: false
+        };
+        this.options = {
+            lineNumbers: true,
+            tabSize: 4,
+            readOnly: false,
+            extraKeys: {},
+            mode: getHighlightMode(this.props.dialect)
+        };
+        this.updateCode = this.updateCode.bind(this);
+        this.handleIntervalChange = this.handleIntervalChange.bind(this);
+        this.handleFilenameChange = this.handleFilenameChange.bind(this);
+        this.submit = this.submit.bind(this);
+    }
+
+    updateCode(editor, meta, code) {
+        this.setState({ code });
+    }
+
+    handleIntervalChange({ value: intervalType }) {
+        this.setState({ intervalType });
+    }
+
+    handleFilenameChange(e) {
+        this.setState({ filename: e.target.value });
+    }
+
+    submit() {
+        if (!this.state.code || !this.state.code.length) {
+            return this.setState({
+                error: 'Please enter the query to be scheduled above.'
+            });
+        }
+        // if (!this.state.filename || !this.state.filename.length) {
```

Comment:
Please, ignore this comment for now. The second condition isn't necessary, `if (!this.state.code)` is enough, because an empty string is converted to false.